internal/controller/snapshot/snapshot_adapter.go

```diff
@@ -433,6 +435,73 @@ func (a *Adapter) EnsureAllReleasesExist() (controller.OperationResult, error) {
 	return controller.ContinueProcessing()
 }
 
+// EnsureOverrideSnapshotValid is an operation that ensure the manually created override snapshot have valid
+// digest and git source in snapshotComponents, mark it as invalid if it is invalid
+func (a *Adapter) EnsureOverrideSnapshotValid() (controller.OperationResult, error) {
+	if !gitops.IsOverrideSnapshot(a.snapshot) {
+		a.logger.Info("The snapshot was not override snapshot, skipping")
+		return controller.ContinueProcessing()
+	}
+
+	if gitops.IsSnapshotMarkedAsInvalid(a.snapshot) {
+		a.logger.Info("The override snapshot has been marked as invalid, skipping")
+		return controller.ContinueProcessing()
+	}
```

Comment:
Let's say:
* A user created an override snapshot with an invalid digest, so we mark it as invalid (as per the logic defined in the for loop somewhere below this LOC)
* Now they realized the reason behind why it is invalid, so they fixed the digest.
* But now this Snapshot is stuck at Invalid state because we call `return controller.ContinueProcessing()` here.

Should we reconsider this `if` condition?

Reply:
Good question. The snapshotComponents in other type of snapshot is immutable. For this type of `override` snapshot, let me think.

Reply:
For snapshot, we reconcile it when it is created or its annotation is updated, if users wants to reconcile it after updating it, they have to create a new snapshot.

Reply:
Thanks for the review, it is not urgent.

Reply:
> if users wants to reconcile it after updating it, they have to create a new snapshot.

that's not a very user-friendly UX, but I'm okay with it.

If we have 100s of 'dumb' users creating override snapshots then we'd have at least 100s of invalid override snapshots that are useless in the cluster. Should we then need to update the garbage collector to delete straightaway the **invalid override** snapshots?

Reply:
Snapshot GC has covered it.